code/xcvm/cosmwasm/contracts/escrow/src/auth.rs

```diff
@@ -126,6 +150,31 @@ pub(crate) fn handle_break_glass(_: Admin, deps: DepsMut, env: Env, _info: Messa
 }
 
 pub(crate) mod policy {
-	pub(crate) enum User {}
-	pub(crate) enum Admin {}
+	use super::*;
+
+	#[derive(Clone, Default)]
+	pub(crate) struct User;
+
+	#[derive(Clone, Default)]
+	pub(crate) struct Admin;
+
+	#[derive(Clone)]
+	pub(crate) struct Cw20Contract {
+		// address: cw20::Cw20Contract,
+		asset_id: xc_core::AssetId,
+	}
+
+	impl Cw20Contract {
+		/// Creates a new `Cw20Contract` if the address corresponds to a known
+		/// asset.
+		///
+		/// If it doesn’t, returns [`ContractError::UnknownAsset`] error.
+		pub fn new(_storage: &dyn Storage, _address: Addr) -> Result<Self> {
+			todo!()
```

Comment:
not sure, how to create contract on address ahead? afaik it is always unpredictalbe

Reply:
As far as I understand, for CW20 we get messages from CW20 contract and need to be able to understand that contract’s address.  The point of asset registry (which the code is currently lacking) is to translate that address to asset id.  So that’s what this method will eventually do.